#1813

latenitefilms opened the pull request:
- Attempted to make sure that stabilisation is enabled, before changing
the stabilisation method. Not currently working as expected.

latenitefilms committed:
#1813

- Attempted to make sure that stabilisation is enabled, before changing
the stabilisation method. Not currently working as expected.

latenitefilms committed:
#1813

- Fixed typo

latenitefilms committed:
#1813

- Fixed bug in `doStabilizationMethod`

randomeizer commented on src/plugins/finalcutpro/inspector/video.lua:
```diff
@@ -144,25 +145,43 @@ end
-        :Then(true)
-    end)
+    end):Is(true):Then(
+        If(stabilization:doShow()):Is(true)
```
The `If` will default to checking if it `Is(true)`, so that can be skipped. i.e.
```suggestion
        If(stabilization:doShow())
```

randomeizer commented on src/plugins/finalcutpro/inspector/video.lua:
```diff
@@ -144,25 +145,43 @@ end
+    end):Is(true):Then(
+        If(stabilization:doShow()):Is(true)
+        :Then(
+            If(function() return stabilization:isShowing() end):Is(true)
```
Same here...
```suggestion
            If(function() return stabilization:isShowing() end)
```

randomeizer replied:
In fact, I'm pretty sure that `isShowing` in this case is a `cp.prop`, so we can just pass it in directly, like so:
```suggestion
            If(stabilization.isShowing)
```

randomeizer replied:
The reason we can do this is that the value of `stabilization.isShowing` will update dynamically. The main trick with Rx is that you generally don't want to resolve the values being passed into the various `If` or `Do` functions immediately - they will get resolved when the final statement is actually executed (which may happen multiple times in its lifetime). `cp.prop` values are always up-to-date, so they can be checked for updates on demand.

randomeizer commented on src/plugins/finalcutpro/inspector/video.lua:
```diff
@@ -144,25 +145,43 @@ end
+            If(function() return stabilization:isShowing() end):Is(true)
+            :Then(
+                Do(
+                    If(function() return stabilization.enabled:value() end):Is(false)
```
This can be:
```suggestion
                    If(stabilization.enabled.value):Is(false)
```

randomeizer commented on src/plugins/finalcutpro/inspector/video.lua:
```diff
@@ -144,25 +145,44 @@ end
+        :Then(
+            If(function() return stabilization:isShowing() end):Is(true)
+            :Then(
+                If(function() return stabilization.enabled:checked() end):Is(false)
```
```suggestion
                If(stabilization.enabled.checked):Is(false)
```

latenitefilms committed:
#1813

- `cp.ui.PopUpButton:doSelectValue()` now no longer shows the popup if
the value that’s been requested is already selected.
- Cleaned up `doStabilizationMethod()`.
- Closes #1813

latenitefilms committed:
#1813

- Fixed @stickler-ci bug

randomeizer commented on src/extensions/cp/ui/PopUpButton.lua:
```diff
@@ -157,19 +159,25 @@ end
-    end)
-    :Then(WaitUntil(self.menuUI):Is(nil):TimeoutAfter(TIMEOUT_AFTER))
+    :Then(
+        If(function() return self.value and self:value() == value end):Then(true)
```
I believe that `self.value` will never be `nil` (it's a lazy prop, so it gets created on demand), so this could just be:

randomeizer replied:
```suggestion
        If(self.value):Is(value):Then(true)
```

randomeizer commented on src/plugins/finalcutpro/inspector/video.lua:
```diff
@@ -160,16 +159,19 @@ local function doStabilizationMethod(value)
+                :Then(WaitUntil(stabilization.enabled):Is(true):TimeoutAfter(2000))
+            )
+            :Then(
+                If(function() return method():UI():enabled() end) -- Only try and "tick" it if it's enabled. The stabilisation might still be processing.
```
The `PopUpButton` subclasses `Element` and inherits its `isEnabled` `cp.prop`, so we can reference that directly:

randomeizer replied:
```suggestion
                If(method.isEnabled)) -- Only try and "tick" it if it's enabled. The stabilisation might still be processing.
```

randomeizer reviewed (commented):
A couple of additional simplifications. Untested...

randomeizer committed:
Update src/extensions/cp/ui/PopUpButton.lua

Co-Authored-By: latenitefilms <latenitefilms@users.noreply.github.com>

randomeizer committed:
Update src/plugins/finalcutpro/inspector/video.lua

Co-Authored-By: latenitefilms <latenitefilms@users.noreply.github.com>

latenitefilms committed:
#1813

- Fixed @stickler-ci bug

latenitefilms committed:
#1813

- Now enables Rolling Shutter if you trigger a Rolling Shutter Amount
action.
- Closes #1813
